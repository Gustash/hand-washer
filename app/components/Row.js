import React from 'react'
import { View, Text, StyleSheet } from 'react-native'
import { colors } from 'app/config'

const Row = ({
  highlight,
  highlightStyle,
  primaryText,
  secondaryText,
  style,
  separator = false,
}) => (
  <View style={[styles.container, separator && styles.separator, style]}>
    <View
      testId="row-highlight-container"
      style={[styles.highlightContainer, highlightStyle]}
    >
      <Text testId="row-highlight-text" style={styles.highlightText}>
        {highlight}
      </Text>
    </View>
    <View style={styles.textContainer}>
      <Text testId="row-primary-text" style={styles.primaryText}>
        {primaryText}
      </Text>
      <Text testId="row-secondary-text" style={styles.secondaryText}>
        {secondaryText}
      </Text>
    </View>
  </View>
)

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    paddingVertical: 16,
  },
  separator: {
    borderBottomColor: colors.separator,
    borderBottomWidth: 1,
  },
  highlightContainer: {
    backgroundColor: colors.green,
    paddingHorizontal: 14,
    paddingVertical: 8,
    width: 80,
    justifyContent: 'center',
    marginHorizontal: 8,
  },
  highlightText: {
    textAlign: 'center',
    color: colors.white,
    fontSize: 17,
  },
  primaryText: {
    fontSize: 17,
    color: colors.darkGrey,
  },
  textContainer: {
    flex: 1,
  },
  secondaryText: {
    fontSize: 14,
    color: colors.lightGrey,
  },
})

export default Row
