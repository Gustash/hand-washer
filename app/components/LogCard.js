import React, { useMemo } from 'react'
import moment from 'moment'

import Row from 'app/components/Row'

const LogCard = ({ date, format = 'DD-MM-YYYY HH:mm', ...props }) => {
  const momentDate = useMemo(() => moment(date, format), [date, format])
  const weekday = useMemo(() => {
    const isoWeekday = momentDate.isoWeekday()

    switch (isoWeekday) {
      case 1:
        return 'Monday'
      case 2:
        return 'Tuesday'
      case 3:
        return 'Wednesday'
      case 4:
        return 'Thursday'
      case 5:
        return 'Friday'
      case 6:
        return 'Saturday'
      case 7:
        return 'Sunday'
      // Should never happen
      default:
        return ''
    }
  }, [momentDate])

  return (
    <Row
      highlight={momentDate.format('HH:mm')}
      primaryText={weekday}
      secondaryText={momentDate.format('Do MMMM YYYY')}
      separator
      {...props}
    />
  )
}

export default LogCard
