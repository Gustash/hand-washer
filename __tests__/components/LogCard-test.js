import React from 'react'
import { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'

import LogCard from '../../app/components/LogCard'
import Row from '../../app/components/Row'

const mockProps = props => ({
  date: '01-01-1990 00:00',
  ...props,
})

const matchesDate = (wrapper, highlight, primaryText, secondaryText) => {
  expect(wrapper.find(Row).props()).toMatchObject({
    highlight,
    primaryText,
    secondaryText,
  })
}

describe('<LogCard />', () => {
  let props
  let wrapper

  beforeEach(() => {
    props = mockProps()
    wrapper = shallow(<LogCard {...props} />)
  })

  it('matches snapshot', () => {
    expect(shallowToJson(wrapper)).toMatchSnapshot()
  })

  it('applies date', () => {
    matchesDate(wrapper, '00:00', 'Monday', '1st January 1990')
    wrapper.setProps({
      ...props,
      date: '02-01-1990 01:10',
    })
    matchesDate(wrapper, '01:10', 'Tuesday', '2nd January 1990')
    wrapper.setProps({
      ...props,
      date: '03-01-1990 02:20',
    })
    matchesDate(wrapper, '02:20', 'Wednesday', '3rd January 1990')
    wrapper.setProps({
      ...props,
      date: '04-01-1990 03:30',
    })
    matchesDate(wrapper, '03:30', 'Thursday', '4th January 1990')
    wrapper.setProps({
      ...props,
      date: '05-01-1990 04:40',
    })
    matchesDate(wrapper, '04:40', 'Friday', '5th January 1990')
    wrapper.setProps({
      ...props,
      date: '06-01-1990 05:50',
    })
    matchesDate(wrapper, '05:50', 'Saturday', '6th January 1990')
    wrapper.setProps({
      ...props,
      date: '07-01-1990 06:13',
    })
    matchesDate(wrapper, '06:13', 'Sunday', '7th January 1990')
    wrapper.setProps({
      ...props,
      date: 'Invalid date',
    })
    matchesDate(wrapper, 'Invalid date', '', 'Invalid date')
  })

  it('uses format', () => {
    wrapper.setProps({
      ...props,
      date: '1990/02/01 01:02:13',
      format: 'YYYY/DD/MM HH:mm:ss',
    })
    matchesDate(wrapper, '01:02', 'Tuesday', '2nd January 1990')
  })

  it('spreads props', () => {
    wrapper.setProps({
      ...props,
      style: {
        marginTop: 1000,
      },
    })
    expect(wrapper.find(Row).props()).toMatchObject({
      style: {
        marginTop: 1000,
      },
    })
  })
})
