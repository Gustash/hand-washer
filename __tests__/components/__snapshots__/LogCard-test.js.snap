// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LogCard /> matches snapshot 1`] = `
<Row
  highlight="00:00"
  primaryText="Monday"
  secondaryText="1st January 1990"
  separator={true}
/>
`;
